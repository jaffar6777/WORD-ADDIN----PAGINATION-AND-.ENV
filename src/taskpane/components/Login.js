import { Label, PrimaryButton, TextField } from "@fluentui/react";
import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { Login as userLogin } from "../store/AuthState/AuthActions";

const container = {
  height: "100vh",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  justifyContent: "center",
};

const m1 = {
  margin: "1rem",
};

const mt1 = {
  marginTop: "1rem",
};

const mb1 = {
  marginBottom: "1rem",
};

function Login() {
  const dispatch = useDispatch();
  const [username, setUsername] = useState("");
  const [usernameError, setUsernameError] = useState(false);
  const [password, setPassword] = useState("");
  const [passwordError, setPasswordError] = useState(false);

  const onChangeUserName = (event) => {
    setUsername(event.target.value);
    setUsernameError(false);
  };

  const onChangePassword = (event) => {
    setPassword(event.target.value);
    setPasswordError(false);
  };

  const submitHandler = () => {
    if (username === "") {
      setUsernameError("Please Enter User Name");
    } else if (password === "") {
      setPasswordError("Please Enter Password");
    } else if (username !== "Admin") {
      setUsernameError("Please check the User Name");
    } else if (password !== "admin") {
      setPasswordError("Please check the Password");
    }
    dispatch(userLogin());
    setUsername("");
    setPassword("");
  };

  return (
    <div style={container}>
      <h2 className={m1}>LOGIN</h2>
      <form onSubmit={submitHandler}>
        <Label required className={mt1}>
          User Name
        </Label>
        <TextField
          placeholder="User Name"
          errorMessage={usernameError ? usernameError : ""}
          onChange={onChangeUserName}
        />
        <Label required>Password</Label>
        <TextField
          className={mb1}
          placeholder="Password"
          errorMessage={passwordError ? passwordError : ""}
          onChange={onChangePassword}
        />
        <PrimaryButton text="Submit" onClick={submitHandler} />
      </form>
    </div>
  );
}

export default Login;
