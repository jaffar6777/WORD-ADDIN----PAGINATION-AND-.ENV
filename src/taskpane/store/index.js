import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";
import { loginReducer } from "./AuthState/AuthReducer";
import fetchReducer from "./PostsState/PostsReducer";

const rootReducer = combineReducers({ Auth: loginReducer, Posts: fetchReducer });

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
