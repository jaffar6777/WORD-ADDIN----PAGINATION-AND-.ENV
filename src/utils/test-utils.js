// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { Provider } from "react-redux";

import { applyMiddleware, combineReducers, createStore } from "redux";
import { loginReducer } from "../taskpane/store/AuthState/AuthReducer";
import postsReducer from "../taskpane/store/PostsState/PostsReducer";
import thunk from "redux-thunk";

const rootReducer = combineReducers({ Auth: loginReducer, Posts: postsReducer });

let reduxTestStore = null;
const render = (
  component,
  { initialState, store = createStore(rootReducer, initialState, applyMiddleware(thunk)), ...renderOptions } = {}
) => {
  reduxTestStore = store;
  const Wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;
  return rtlRender(component, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from "@testing-library/react";
// override render method
export { render, reduxTestStore };
