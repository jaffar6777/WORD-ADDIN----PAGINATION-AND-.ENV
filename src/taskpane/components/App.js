import React from "react";
import { useSelector } from "react-redux";
import Home from "./Home";
import Login from "./Login";

function App() {
  const isLoggedIn = useSelector((state) => state.Auth.isLoggedIn);
  return <div className="root">{isLoggedIn ? <Home /> : <Login />}</div>;
}

export default App;
