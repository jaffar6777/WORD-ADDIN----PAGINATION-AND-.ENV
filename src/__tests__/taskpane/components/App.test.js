import App from "../../../taskpane/components/App";
import { render } from "../../../utils/test-utils";
import React from "react";
import TestRenderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../../taskpane/store";
import Login from "../../../taskpane/components/Login";

const MockedLoginComponent = () => (
  <Provider store={store}>
    <Login />
  </Provider>
);

describe("App", () => {
  it("should render Login when user is not Logged in", async () => {
    render(<App />);
    const snapShot = TestRenderer.create(<MockedLoginComponent />).toJSON();
    expect(snapShot).toMatchSnapshot();
  });
});
