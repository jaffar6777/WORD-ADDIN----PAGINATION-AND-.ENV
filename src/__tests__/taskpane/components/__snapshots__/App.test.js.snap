// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render Login when user is not Logged in 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
      "justifyContent": "center",
    }
  }
>
  <h2
    className={
      Object {
        "margin": "1rem",
      }
    }
  >
    LOGIN
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      className="ms-Label root-53"
    >
      User Name
    </label>
    <div
      className="ms-TextField root-55"
    >
      <div
        className="ms-TextField-wrapper"
      >
        <div
          className="ms-TextField-fieldGroup fieldGroup-56"
        >
          <input
            aria-invalid={false}
            className="ms-TextField-field field-57"
            id="TextField9"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            placeholder="User Name"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <label
      className="ms-Label root-66"
    >
      Password
    </label>
    <div
      className="ms-TextField root-67"
    >
      <div
        className="ms-TextField-wrapper"
      >
        <div
          className="ms-TextField-fieldGroup fieldGroup-56"
        >
          <input
            aria-invalid={false}
            className="ms-TextField-field field-57"
            id="TextField12"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            placeholder="Password"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <button
      className="ms-Button ms-Button--primary root-68"
      data-is-focusable={true}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      <span
        className="ms-Button-flexContainer flexContainer-69"
        data-automationid="splitbuttonprimary"
      >
        <span
          className="ms-Button-textContainer textContainer-70"
        >
          <span
            className="ms-Button-label label-72"
            id="id__15"
          >
            Submit
          </span>
        </span>
      </span>
    </button>
  </form>
</div>
`;
